### ISSUE TITLE
Unclosed client session when using SPO connector

### ISSUE THREAD

**Issue by artem-shelkovnikov** [elastic/connectors-python#1147]
## Bug Description
When running Sharepoint Online connector, errors constantly appear in logs:

```
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x108eeca30>
```

### To Reproduce
1. Configure an on-prem Sharepoint Online connector
2. Run connector service (no need to run a sync)
3. Observe these warnings appear in logs

## Expected behavior
No warnings appear in logs about unclosed session

**Comment by navarone-feekery** [comment]
@artem-shelkovnikov I had a look at this from a data extraction service perspective. I commented out every instance that a client session is created, but the log continued to show. I don't think this is related to the data extraction service.

I then looked through the sharepoint online connector and commented out the http session here: https://github.com/elastic/connectors-python/blob/f584d4ff0ad82d5aba42c1fac48bbb481e067c84/connectors/sources/sharepoint_online.py#L384-L391 I re-ran everything and the log disappeared.

I'm not sure how to fix this, and because I have ongoing feature work I won't have scope until maybe next week. Could I pass this on to you? (or someone else you think might be knowledgeable about this part of the codebase).
